internal/objects/objects.go

```diff
@@ -29,14 +29,7 @@ import (
 	nropv1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1"
 )
 
-func NewNUMAResourcesOperator(name string, labelSelectors []*metav1.LabelSelector) *nropv1.NUMAResourcesOperator {
-	var nodeGroups []nropv1.NodeGroup
-	for _, selector := range labelSelectors {
-		nodeGroups = append(nodeGroups, nropv1.NodeGroup{
-			MachineConfigPoolSelector: selector,
-		})
-	}
-
+func NewNUMAResourcesOperator(name string, nodeGroups []nropv1.NodeGroup) *nropv1.NUMAResourcesOperator {
```

Comment:
since this is an internal API, and strictly while this is an internal API, what about
```
func NewNUMAResourcesOperator(name string, nodeGroups ...nropv1.NodeGroup) *nropv1.NUMAResourcesOperator {
// ...
```